### PR TITLE
Mitm manager factory

### PR DESCRIPTION
**Pull request by viacheslav-fomin-main**
Follow up to: https://github.com/verygoodsecurity/vault/pull/1380

Introduces mitm manager factory which can create a different mitm manager based on channel attributes.

### CODE REVIEW

File: src/main/java/org/littleshoot/proxy/MitmManagerFactory.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+public interface MitmManagerFactory {
+
+  /**
+   * Retrieves an instance of {@link MitmManager} based on specific attributes from cxt channel
+   *
+   * @param channel current channel from cxt
+   * @return concrete instance of {@link MitmManager}
+   */
+  MitmManager getInstance(Channel channel);
+}
```
Comment (viacheslav-fomin-main) [inline src/main/java/org/littleshoot/proxy/MitmManagerFactory.java:14]: here is how we can check tenant preferences: `channel.attr(ProxyHttpFiltersAdapter.preferences).get().getTrustAllServers()`

File: src/main/java/org/littleshoot/proxy/impl/ProxyToServerConnection.java [hunk 2]
```diff
@@ -556,7 +556,7 @@ private void initializeConnectionFlow() {
                         serverConnection.HTTPCONNECTWithChainedProxy);
```
Comment (viacheslav-fomin-main) [inline src/main/java/org/littleshoot/proxy/impl/ProxyToServerConnection.java:556]: Tenant related attributes are stored in client to proxy connection channel

File: src/main/java/org/littleshoot/proxy/MitmManagerFactory.java (hunk 1 above)
Comment (viacheslav-fomin-main) [reply src/main/java/org/littleshoot/proxy/MitmManagerFactory.java:14]: I do not really like exposing the whole `channel` here but there is no easy way to get all the attributes at once from the channel

File: src/test/java/org/littleshoot/proxy/MITMUsernamePasswordAuthenticatingProxyTest.java [hunk 3]
```diff
@@ -14,7 +15,12 @@ protected void setUp() {
         this.proxyServer = bootstrapProxy()
                 .withPort(0)
                 .withProxyAuthenticator(this)
-                .withManInTheMiddle(new SelfSignedMitmManager())
+                .withManInTheMiddle(new MitmManagerFactory() {
```
Comment (mjallday) [inline src/test/java/org/littleshoot/proxy/MITMUsernamePasswordAuthenticatingProxyTest.java]: can we do `SelfSignedMitmManagerFactory` here?

**Review by mjallday — APPROVED** [review]

File: src/test/java/org/littleshoot/proxy/MITMUsernamePasswordAuthenticatingProxyTest.java (hunk 3 above)
Comment (viacheslav-fomin-main) [reply src/test/java/org/littleshoot/proxy/MITMUsernamePasswordAuthenticatingProxyTest.java]: sure, will fix, thanks!

**Pull request merged**